Tests/KakapoDBTests.swift

```diff
@@ -271,31 +274,39 @@ class KakapoDBTests: QuickSpec {
                 
                 let elementToDelete = UserFactory(firstName: "Joan", lastName: "Romano", age: 28, id: theId)
                 try! sut.delete(elementToDelete)
-                let usersArray = sut.findAll(UserFactory.self)
+                let usersArray = sut.findAll(UserFactory)
                 
                 expect(usersArray.count).to(equal(0))
             }
             
-            it("should not delete a non previously inserted object, even with same ids, since they have different data representation") {
-                sut.create(UserFactory.self, number: 20)
-                let elementToDelete = UserFactory(id: 2, db: sut)
+            it("should not delete an object with same id but different type") {
+                let userFactory = sut.create(UserFactory).first!
+                let elementToDelete = CommentFactory(text: "", likes: 0, id: userFactory.id)
                 expect{ try sut.delete(elementToDelete) }.to(throwError(errorType: KakapoDBError.self))
-                expect(sut.findAll(UserFactory.self).count).to(equal(20))
+                expect(sut.findAll(UserFactory).count).to(equal(1))
+            }
+            
+            it("should delete an object with same id and different type but different properties") {
```

Comment:
shouldn't this be `should delete an object with *same id and type* but different properties` ?